### DISCUSSION TITLE
Add compatible `Collection` functionality to `DataCollection`

### DISCUSSION THREAD

**Discussion by andrey-helldar** [spatie/laravel-data#635] [Ideas]
Right now, the DataCollection object has very poor functionality, which severely limits its scope.

For example, it is impossible to group after creation. That's why we have to use this trick:

```php
function countriesList(DataCollection $items): Collection
{
    return collect($items->all())->groupBy('type')->map(
        fn (Collection $values) => $values->pluck('countries')->flatten()->all()
    );
}

return countriesList(
    SomeData::collection($items)
);
```
Instead of doing this:
```php
return SomeData::collection($items)->groupBy('type')->map(
    fn (Collection $values) => $values->pluck('countries')->flatten()->all()
);
```

It would be cool to have the same functionality in a DataCollection object as Laravel's Collection.

PS: You also can't add macros.

**Comment by czuniga9** [comment]
Can you use `$items->toCollection()->groupBy...`?

**Reply by andrey-helldar** [reply]
Wow. I somehow missed this method...
